### PR TITLE
aws-sdk: Provides its own types

### LINKED ISSUE #7458: Definitions for aws-sdk are woefully out of date
The definitions are missing a bunch of services and aren't even close to being up-to-date to the sdk's API.

@enicholson and @aldrinleal have done a bunch of work creating a generator in the [ingenieux/aws-sdk-typescript](https://github.com/ingenieux/aws-sdk-typescript) repo. Is there any way we could get the output of it merged into this repo so we can use `tsd`?

### PR DESCRIPTION
**Pull request by ghost**
Fixes #7458 and #10356.
This will deprecate `@types/aws-sdk` in favor of just `aws-sdk`. CC @midknight41, @CasperSkydt

### CODE REVIEW

**Comment by westy92** [comment]
Looks good!

File: types/nodemailer-ses-transport/index.d.ts [hunk 1]
```diff
@@ -3,24 +3,23 @@
 // Definitions by: Seth Westphal <https://github.com/westy92/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare module "nodemailer-ses-transport" {
-	import * as AWS from 'aws-sdk';
-	import * as nodemailer from "nodemailer";
+import * as AWS from 'aws-sdk';
```
Comment (westy92) [inline types/nodemailer-ses-transport/index.d.ts]: Should we make quote usage consistent?  (`"` vs. `'`)
https://palantir.github.io/tslint/rules/quotemark/

**Comment by midknight41** [comment]
👍

**Comment by BendingBender** [comment]
One of the packages to remove from the list in #13364

**Pull request merged**